commercetools-models/src/test/java/io/sphere/sdk/products/commands/ProductUpdateCommandIntegrationTest.java

```diff
@@ -944,46 +1297,53 @@ public void changeMasterVariantWithSku() {
         });
     }
 
-    private static class StagedWrapper extends Base implements UpdateAction<Product> {
-        private final UpdateAction<Product> delegate;
-        @JsonProperty("staged")
-        private final boolean staged;
+    @Test
+    public void changeMasterVariantWithSkuWithStaged() {
+        changeMasterVariantWithSkuWithStaged(true);
+        changeMasterVariantWithSkuWithStaged(false);
+    }
+
+    private void changeMasterVariantWithSkuWithStaged(final boolean staged) {
+        withUpdateableProductOfMultipleVariants(client(), product -> {
+            assertThat(product.getMasterData().hasStagedChanges()).isFalse();
 
-        private StagedWrapper(final UpdateAction<Product> action, final boolean staged) {
-            this.delegate = action;
-            this.staged = staged;
-        }
+            final int originalMasterVariantId = product.getMasterData().getStaged().getMasterVariant().getId();
+            final ProductVariant variantSupposedToBeMaster = product.getMasterData().getStaged().getAllVariants().get(2);
+            final String sku = variantSupposedToBeMaster.getSku();
 
-        @Override
-        public String getAction() {
-            return delegate.getAction();
-        }
+            final ChangeMasterVariant updateAction = ChangeMasterVariant.ofSku(sku, staged);
+            final Product updatedProduct = client().executeBlocking(ProductUpdateCommand.of(product, updateAction));
 
-        @JsonUnwrapped
-        public UpdateAction<Product> getDelegate() {
-            return delegate;
-        }
+            assertThat(updatedProduct.getMasterData().hasStagedChanges()).isEqualTo(staged);
+            assertThat(updatedProduct.getMasterData().getStaged().getMasterVariant().getSku()).isEqualTo(sku);
+            assertThat(updatedProduct.getMasterData().getStaged().getMasterVariant().getId())
+                    .as("variant IDs don't change in reordering")
+                    .isNotEqualTo(originalMasterVariantId)
+                    .isEqualTo(variantSupposedToBeMaster.getId());
+
+            return updatedProduct;
+        });
     }
 
     @Test
     public void possibleToHackUpdateForStagedAndCurrent() throws Exception {
```

Comment:
Is this test really still necessary? Does it test a different test case then the test that you created? If it covers a different test case then please rename. Otherwise please just remove this test.